orange-pi/5-max/kernel/default.nix

```diff
@@ -0,0 +1,26 @@
+{ fetchFromGitHub
+, lib
+, linuxManualConfig
+, stdenv
+, ubootTools
+, ...
+}:
+(linuxManualConfig rec {
+  inherit lib stdenv;
+  modDirVersion = "6.1.75";
+  version = "${modDirVersion}-rk3588";
+  src = fetchFromGitHub {
+    owner = "armbian";
+    repo = "linux-rockchip";
+    rev = "v24.11.1";
+    hash = "sha256-ZqEKQyFeE0UXN+tY8uAGrKgi9mXEp6s5WGyjVuxmuyM=";
+  };
+  # fetched from https://github.com/armbian/build/blob/v24.11.1/config/kernel/linux-rk35xx-vendor.config
+  configfile = ./linux-rk35xx-vendor.config;
+  # converted to nix using https://github.com/NixOS/nixpkgs/blob/79aaddff29307748c351a13d66f9d1fba4218624/pkgs/os-specific/linux/kernel/manual-config.nix#L11-L19
+  config = import ./linux-rk35xx-vendor.nix;
```

Comment:
It's a bit unfortunate that NixOS requires these settings to be present to. Merging this pull request would double the size of this repository in terms of lines of code. On the one hand it would be great to have this kernel in but on the other it adds quite a bit of weight. Do you know where we use this information in NixOS?

Reply:
What would happen if we provide the config of a similar kernel version? Is this just about safety checks?

Reply:
I did test without these configs set, but I get failed kernel builds. But looks like since I made this pr kernel 6.14 is now the latest in nixos-unstable and has mainline support for orangepi 5 max. Going to test if we can just use latest mainline instead 👀

Reply:
That would make it a lot more usable for everyone.